### PR TITLE
Add stETH wrap/unwrap support [TKR-377]

### PR DESCRIPTION
**Pull request by kh-chang**
## Description

Add stETH wrap/unwrap support

## Types of changes

* New feature (non-breaking change which adds functionality) 

## Checklist:

-   [x] Prefix PR title with `[WIP]` if necessary.
-   [x] Add tests to cover changes as needed.
-   [x] Update documentation as needed.
-   [x] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: contracts/zero-ex/contracts/src/transformers/bridges/mixins/MixinLido.sol [hunk 1]
```diff
@@ -59,12 +86,43 @@ contract MixinLido {
...
+
+        return _tradeWstETH(sellToken, buyToken, sellAmount, bridgeData);
+    }
+
+    function _tradeStETH(
+        IERC20TokenV06 buyToken,
+        uint256 sellAmount,
+        bytes memory bridgeData
+    ) private returns (uint256 boughtAmount) {
+        (IStETH stETH) = abi.decode(bridgeData, (IStETH));
```
Comment (kh-chang) [inline contracts/zero-ex/contracts/src/transformers/bridges/mixins/MixinLido.sol:101]: @dekz I tried to make the contract backward compatible. 

With the change in PR `bridgeData` will always contain two addresses (stETH and wstETH) but I think that's fine.

File: packages/asset-swapper/src/utils/market_operation_utils/constants.ts [hunk 2]
```diff
@@ -2478,7 +2486,17 @@ export const DEFAULT_GAS_SCHEDULE: Required<FeeSchedule> = {
...
+    [ERC20BridgeSource.Lido]: (fillData?: FillData) => {
+        const lidoFillData = fillData as LidoFillData;
+        const wethAddress = NATIVE_FEE_TOKEN_BY_CHAIN_ID[ChainId.Mainnet];
+        // WETH -> stETH
+        if (lidoFillData.takerToken === wethAddress) {
+            return 226e3;
+        }
+
+        // Wrapping or unwraping stETH.
+        return lidoFillData.takerToken === lidoFillData.stEthTokenAddress ? 120e3 : 95e3;
```
Comment (mzhu25) [inline packages/asset-swapper/src/utils/market_operation_utils/constants.ts:2498]: can we turn this into an `if {} else if {} else {}`?

File: contracts/zero-ex/contracts/src/transformers/bridges/mixins/MixinLido.sol [hunk 3]
```diff
@@ -59,12 +86,43 @@ contract MixinLido {
...
+        IERC20TokenV06 buyToken,
+        uint256 sellAmount,
+        bytes memory bridgeData
+
+    ) private returns(uint256 boughtAmount){
+        (IEtherTokenV06 stETH, IWstETH wstETH) = abi.decode(bridgeData, (IEtherTokenV06, IWstETH));
+        sellToken.approveIfBelow(address(wstETH), sellAmount);
+        if (address(sellToken) == address(stETH) && address(buyToken) == address(wstETH) ) {
+            return wstETH.wrap(sellAmount);
+        }
```
Comment (mzhu25) [inline contracts/zero-ex/contracts/src/transformers/bridges/mixins/MixinLido.sol:121]:
```suggestion
        if (address(sellToken) == address(stETH) && address(buyToken) == address(wstETH) ) {
            sellToken.approveIfBelow(address(wstETH), sellAmount);
            return wstETH.wrap(sellAmount);
        }
```

**Review by mzhu25 — APPROVED** [review]
nice work!

File: contracts/zero-ex/contracts/src/transformers/bridges/mixins/MixinLido.sol (hunk 3 above)
Comment (kh-chang) [reply contracts/zero-ex/contracts/src/transformers/bridges/mixins/MixinLido.sol]: ah, thanks for catching!

File: packages/asset-swapper/contracts/src/LidoSampler.sol [hunk 4]
```diff
@@ -72,20 +94,35 @@ contract LidoSampler is SamplerUtils {
...
 
-        // Minting stETH is always 1:1 therefore we can just return the same amounts back
-        return makerTokenAmounts;
-    }
+        if (takerToken == lidoInfo.wstEthToken && makerToken == lidoInfo.stEthToken) {
+            for (uint256 i = 0; i < numSamples; i++) {
+                takerTokenAmounts[i] = wstETH.getWstETHByStETH(makerTokenAmounts[i]);
+            }
+            return takerTokenAmounts;
+        }
```
Comment (dekz) [inline packages/asset-swapper/contracts/src/LidoSampler.sol:123]: On the surface these look like the same functions used in sells and buys? 

```
wstETH -> stETH ? wstETH.getWstETHByStETH
stETH -> wstETH ? wstETH.getStETHByWstETH
```

Probably a smaller bytecode footprint if you convert these to internal and reuse them.

**Review by dekz — APPROVED** [review]
So we will need to re-deploy the FQT/BridgeAdapter before this goes out to prod, else we might start giving out `stETH->wstETH` trades we can't settle.

File: packages/asset-swapper/contracts/src/LidoSampler.sol (hunk 4 above)
Comment (kh-chang) [reply packages/asset-swapper/contracts/src/LidoSampler.sol]: very good point! -- I just re-used public sell function in buy. 
Could you take another look?

**Pull request merged**